sdk/go/pulumi/errors/errors.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import "fmt"
+
+type InputPropertyErrorDetails struct {
```

Comment:
I think it'd be good to doc-comment the things we're adding to the public API, though equally happy to wait on that if we expect a bit more discussion on what the API should look like.